### PR TITLE
Coursel

### PR DESCRIPTION
**Pull request by Estif017**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -5,8 +5,9 @@ import {
 	HomePage,
 	SavedPage,
 	UserPage,
-	SearchResults,
+	SearchResultsPage,
```
Comment (Estif017) [inline src/App.js:8]: Changed the name of  SearchResults to SearchResultsPage

File: src/Components/SavedPhotos/SavedPhotos.js [hunk 2]
```diff
@@ -1,28 +1,26 @@
...
-	Image,
-	Button,
-} from './SavedPhotos.styles';
+import Masonry, { ResponsiveMasonry } from 'react-responsive-masonry';
+import { ImageContainer, Image, Button } from './SavedPhotos.styles';
 
 export class SavedPhotos extends Component {
 	render() {
 		const { savedPhotos } = this.props;
+		!savedPhotos.length && <h1>There is no saved photos at the moment</h1>;
```
Comment (yoloOnTheBattlefield) [inline src/Components/SavedPhotos/SavedPhotos.js:8]: this isn't doing anything

File: src/Components/SearchCollections/SearchCollections.js [hunk 3]
```diff
@@ -54,34 +45,18 @@ class SearchCollections extends Component {
 	}
 	render() {
 		const { collections, isLoading, hasError } = this.state;
+		isLoading && <h1>Loading ....</h1>;
+		hasError && <h1>Error ....</h1>;
```
Comment (yoloOnTheBattlefield) [inline src/Components/SearchCollections/SearchCollections.js:49]: same here, nothing happens here

File: src/Components/SearchPhotos/SearchPhotos.js [hunk 4]
```diff
@@ -49,34 +41,16 @@ class SearchPhotos extends Component {
 
 	render() {
 		const { photos, isLoading, hasError } = this.state;
+		isLoading && !hasError && <h1>Loading......</h1>;
+		hasError && !isLoading && <h1>Error......</h1>;
```
Comment (yoloOnTheBattlefield) [inline src/Components/SearchPhotos/SearchPhotos.js:45]: same here

File: src/Components/UserPost/UserPost.js [hunk 5]
```diff
@@ -56,31 +49,13 @@ class UserPost extends Component {
 		isLoading && <h1>Loading ...</h1>;
 		hasError && <h1>Error Occurred</h1>;
```
Comment (yoloOnTheBattlefield) [inline src/Components/UserPost/UserPost.js:50]: same here

File: src/Pages/CollectionsPage/CollectionsPage.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+			this.getPhoto();
+		}
+	}
+	componentDidMount() {
+		this.getPhoto();
+	}
+	render() {
+		const { collections, isLoading, hasError } = this.state;
+		isLoading && <h1>Loading ....</h1>;
+		hasError && <h1>Error ....</h1>;
```
Comment (yoloOnTheBattlefield) [inline src/Pages/CollectionsPage/CollectionsPage.js:46]: same here

File: src/Pages/HomePage/HomePage.js [hunk 7]
```diff
@@ -42,26 +43,25 @@ export default class HomePage extends React.Component {
 	}
 	render() {
 		const { photos, isLoading, hasError } = this.state;
+		isLoading && <h1>Loading ....</h1>;
+		hasError && <h1>Error ....</h1>;
```
Comment (yoloOnTheBattlefield) [inline src/Pages/HomePage/HomePage.js:47]: nothing happens here

File: src/Pages/HomePage/HomePage.js [hunk 8]
```diff
@@ -42,26 +43,25 @@ export default class HomePage extends React.Component {
...
+					<Highlight />
+					{photos.map((photo) => (
+						<Post
+							key={photo.id}
+							photo={photo}
+							addToPhotos={this.props.addToPhotos}
+						/>
+					))}
+				</HomePageContainer>
+			</InfiniteScroll>
```
Comment (yoloOnTheBattlefield) [inline src/Pages/HomePage/HomePage.js:64]:
```return (
  <>
    {isLoading && <h1>Loading ....</h1>}
    {hasError && <h1>Error ....</h1>}
    <InfiniteScroll
      dataLength={this.state.photos.length}
      next={this.fetchNextPage}
      hasMore={true}
      loader={<h4>Loading...</h4>}
    >
      <HomePageContainer>
        <Highlight />
        {photos.map((photo) => (
          <Post
            key={photo.id}
            photo={photo}
            addToPhotos={this.props.addToPhotos}
          />
        ))}
      </HomePageContainer>
    </InfiniteScroll>
  </>
);
```

File: src/Pages/SearchPhotoCollections/SearchPhotoCollections.js [hunk 9]
```diff
@@ -48,35 +41,16 @@ export default class SearchPhotoCollections extends Component {
 	}
 	render() {
 		const { photos, isLoading, hasError, hasMore } = this.state;
-
+		isLoading && <h1>Loading ...</h1>;
+		hasError && <h1>Error Occurred</h1>;
```
Comment (yoloOnTheBattlefield) [inline src/Pages/SearchPhotoCollections/SearchPhotoCollections.js:45]: check above

**Review by yoloOnTheBattlefield — COMMENTED** [review]
check my comments

**Review by yoloOnTheBattlefield — APPROVED** [review]

**Pull request merged**